volttron/platform/certs.py

```diff
@@ -358,18 +359,72 @@ def ca_exists(self):
         """
         return os.path.exists(self.cert_file(self.root_ca_name))
 
+    def get_cert_subject(self, name):
+        """
+        Retrieves the subject details of a certificate
+        :param name: name of the certificate
+        :return: dictionary object with the format
+        {
+            'country':value,
+            'state': value,
+            'location': value,
+            'organization': value,
+            'organization-unit': value,
+            'common-name':value
+        }
+        """
+        subject = self.cert(name).subject
+        return {
+            'common-name': subject.get_attributes_for_oid(
+                NameOID.COMMON_NAME)[0].value,
+            'country': subject.get_attributes_for_oid(
+                NameOID.COUNTRY_NAME)[0].value,
+            'state':subject.get_attributes_for_oid(
+                NameOID.STATE_OR_PROVINCE_NAME)[0].value,
+            'location': subject.get_attributes_for_oid(
+                NameOID.LOCALITY_NAME)[0].value,
+            'organization': subject.get_attributes_for_oid(
+                NameOID.ORGANIZATION_NAME)[0].value,
+            'organization-unit': subject.get_attributes_for_oid(
+                NameOID.ORGANIZATIONAL_UNIT_NAME)[0].value
+        }
+
+
     @staticmethod
     def get_cert_names(instance_name):
```

Comment:
this is a bad name for a function if the doc string is correct.  which based upon what I read it is....I thought based upon the name it would give me all of the certs in the platform.

Reply:
Changed